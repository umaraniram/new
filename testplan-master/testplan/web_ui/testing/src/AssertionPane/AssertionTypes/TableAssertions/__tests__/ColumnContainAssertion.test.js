import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import ColumnContainAssertion from '../ColumnContainAssertion';

function defaultProps() {
  return {
    assertion:{
      "category":"DEFAULT",
      "machine_time":"2019-02-12T17:41:43.215324+00:00",
      "description":null,
      "column":"symbol",
      "type":"ColumnContain",
      "line_no":454,
      "report_fails_only":false,
      "meta_type":"assertion",
      "limit":null,
      "passed":false,
      "values":["AAPL","AMZN"],
      "data":[
        [0,"AAPL",true],
        [1,"GOOG",false],
        [2,"FB",false],
        [3,"AMZN",true],
        [4,"MSFT",false]
      ],
      "utc_time":"2019-02-12T17:41:43.215318+00:00"
    }
  };
}


describe('DictLogAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<ColumnContainAssertion {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
});