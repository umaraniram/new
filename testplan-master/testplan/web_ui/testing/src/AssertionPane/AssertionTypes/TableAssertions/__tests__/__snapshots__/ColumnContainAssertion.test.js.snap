// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictLogAssertion shallow renders the correct HTML structure 1`] = `
<TableBaseAssertion
  columns={
    Array [
      Object {
        "cellStyle": [Function],
        "field": "id",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "width": "72px",
        },
        "title": "ID",
      },
      Object {
        "cellStyle": [Function],
        "field": "value",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "title": "symbol",
      },
    ]
  }
  preText={
    <React.Fragment>
      Values: [
      "AAPL"
      , 
      "AMZN"
      ]
    </React.Fragment>
  }
  rows={
    Array [
      Object {
        "expected": Array [
          "AAPL",
          "AMZN",
        ],
        "id": 0,
        "passed": true,
        "value": "AAPL",
      },
      Object {
        "expected": Array [
          "AAPL",
          "AMZN",
        ],
        "id": 1,
        "passed": false,
        "value": "GOOG",
      },
      Object {
        "expected": Array [
          "AAPL",
          "AMZN",
        ],
        "id": 2,
        "passed": false,
        "value": "FB",
      },
      Object {
        "expected": Array [
          "AAPL",
          "AMZN",
        ],
        "id": 3,
        "passed": true,
        "value": "AMZN",
      },
      Object {
        "expected": Array [
          "AAPL",
          "AMZN",
        ],
        "id": 4,
        "passed": false,
        "value": "MSFT",
      },
    ]
  }
/>
`;
